### DISCUSSION TITLE
How to delete graph from vineyard

### DISCUSSION THREAD

**Discussion by fresher1chen** [alibaba/GraphScope#1317] [Q&A] [answered]
![1644821009365_7DB1452E-E65C-47d4-8D51-9E3CB2C6F6E5](https://user-images.githubusercontent.com/44836806/153816879-b0108d2d-818e-48b9-b1c2-6ad9f53405dc.png)
我想在vineyard 里把 载进去的图 删除掉，但是出现这个问题，请问有什么 解决办法嘛？
![1644821009365_7DB1452E-E65C-47d4-8D51-9E3CB2C6F6E5](https://user-images.githubusercontent.com/44836806/153817292-9057942a-b036-4777-868d-0d3f7afc91fd.png)
我看源码里也是这样写的。

**Comment by sighingnow** [comment]
The `Graph` object in graphscope has a `.unload()` method,

https://github.com/alibaba/GraphScope/blob/main/python/graphscope/framework/graph.py#L605

**Reply by fresher1chen** [reply]
> The `Graph` object in graphscope has a `.unload()` method,
> 
> https://github.com/alibaba/GraphScope/blob/main/python/graphscope/framework/graph.py#L605


您好，按您说的这个方法试了以后，我发现 内存的使用并没有很好的降下去。
我可能没表述清楚，我的场景是这样的，
我载图后，通过pregel 算法 得到一个新的关系，需要创建一个新图。但是用的 本机模式，导致新图 因为 内存不足，无法载图，我想的是把 第一次载的图 占用的内存 释放掉，请问能支持这种操作吗。

**Reply by sighingnow** [reply]
> 我发现 内存的使用并没有很好的降下去。

请问有可以reproduce问题的case吗？

**Reply by fresher1chen** [reply]
import graphscope
sess = graphscope.session(cluster_type="hosts", num_workers=1,vineyard_shared_mem="240G")
sess.as_default()
g = sess.g()
g= g.add_vertices("./big_date.csv",label="xxx")
g.unload()

通过 free -g 观察，g.unload() 语句执行前后，内存 used 没有变化，并没有释放内存。

**Comment by fresher1chen** [comment]
没人关注了吗

**Reply by sighingnow** [reply]
> 没人关注了吗

有的，我reproduce看看哈

**Comment by sighingnow** [comment, marked as answer]
> import graphscope sess = graphscope.session(cluster_type="hosts", num_workers=1,vineyard_shared_mem="240G") sess.as_default() g = sess.g() g= g.add_vertices("./big_date.csv",label="xxx") g.unload()
> 
> 通过 free -g 观察，g.unload() 语句执行前后，内存 used 没有变化，并没有释放内存。

Not reproducible.

`free -g` cannot always reflect the **actual** memory usage of programs, due to the complexity of modern operating system and memory allocator. Instead, you could monitor the shared memory usage of vineyard and how the usage changes before and after `g.unload` as follows,

1. Loading a graph

```python
import graphscope
from graphscope.dataset import load_ogbn_mag
g = load_ogbn_mag()
```

2. Check the shared memory usage of vineyard

```python
import vineyard
vineyard_client = vineyard.connect(g.session.info['engine_config']['vineyard_socket'])
vineyard_client.status
```

Note the `memory_usage` field should be a non-zero value.

3. Unload the graph

```python
g.unload()
```

4. Check the memory usage again

```python
vineyard_client.status
```

The `memory_usage` field return back to 0.
